src/System.Device.Gpio.Tests/RaspberryPi4DriverTests.Linux.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Gpio.Drivers;
+using System.Threading;
+using Xunit;
+
+namespace System.Device.Gpio.Tests
+{
+    public class RaspberryPi4DriverTests : GpioControllerTestBase
+    {
+        /// <summary>
+        /// Leave this pin open (unconnected) for the tests
+        /// </summary>
+        private const int OpenPin = 23;
+
+        protected override GpioDriver GetTestDriver()
+        {
+            return new RaspberryPi4Driver();
+        }
+
+        protected override PinNumberingScheme GetTestNumberingScheme() => PinNumberingScheme.Logical;
+
+        [Fact]
+        public void InputPullResistorsWork()
+        {
+            // This is a hack to test whether we're running on a Pi4 without duplicating that detection logic. 
+            var bestDriver = GpioController.GetBestDriverForBoard();
```

Comment:
I'm slightly skeptical about this check. If our detection logic is broken this will pass because it will not test anything. Can you at least make it so that test is skipped instead so it's more likely to be spotted?

In corefx we have [ConditionalFact](https://github.com/dotnet/corefx/search?q=ConditionalFact&unscoped_q=ConditionalFact) and [SkipTestException](https://github.com/dotnet/corefx/search?q=SkipTestException&unscoped_q=SkipTestException) for that but you can also use xunit Trait if not sure how to use that (i.e. `[Trait("SkipOnCi", "true")]`). For traits you will need to pass something similar to let our CI know to not run it, so i.e. https://github.com/dotnet/iot/blob/master/eng/sendToHelix.proj#L48 except do it unconditionally